src/main/java/appeng/parts/reporting/ConversionMonitorPart.java

```diff
@@ -186,23 +226,65 @@ private void extractItem(Player player, int count) {
 
             var retrieved = StorageHelper.poweredExtraction(energy, cell, itemKey, count,
                     new PlayerSource(player, this));
+
             if (retrieved != 0) {
-                ItemStack newItems = itemKey.toStack((int) retrieved);
+                var newItems = itemKey.toStack((int) retrieved);
+
                 if (!player.getInventory().add(newItems)) {
                     player.drop(newItems, false);
                 }
 
-                if (player.containerMenu != null) {
-                    player.containerMenu.broadcastChanges();
-                }
+                player.containerMenu.broadcastChanges();
             }
         });
     }
 
+    @Override
+    protected void configureWatchers() {
+        // Determine whether our configured item is craftable for requesting crafting jobs via the monitor
+        if (craftingWatcher != null) {
+            craftingWatcher.reset();
+        }
+
+        if (getDisplayed() != null) {
+            if (craftingWatcher != null) {
+                craftingWatcher.add(getDisplayed());
+            }
+        }
+
+        // Update base storage watcher and reported item amount via super call
+        super.configureWatchers();
+    }
+
     @Override
     public IPartModel getStaticModels() {
         return this.selectModel(MODELS_OFF, MODELS_ON, MODELS_HAS_CHANNEL, MODELS_LOCKED_OFF, MODELS_LOCKED_ON,
                 MODELS_LOCKED_HAS_CHANNEL);
     }
 
+    @Override
+    public void returnToMainMenu(Player player, ISubMenu subMenu) {
```

Comment:
Is the menu correctly closed on cancel? Otherwise calling `player.closeContainer` should work.